posthog/api/test/test_dashboard.py

```diff
@@ -8,13 +8,32 @@
 
 from posthog.models import Dashboard, Filter, Insight, Team, User
 from posthog.models.organization import Organization
-from posthog.test.base import APIBaseTest
+from posthog.test.base import APIBaseTest, QueryMatchingTest, snapshot_postgres_queries
 from posthog.utils import generate_cache_key
 
 
-class TestDashboard(APIBaseTest):
+class TestDashboard(APIBaseTest, QueryMatchingTest):
     CLASS_DATA_LEVEL_SETUP = False
 
+    @snapshot_postgres_queries
+    def test_retrieve_dashboard_list(self):
+        d1 = Dashboard.objects.create(team=self.team, name="a dashboard", created_by=self.user)
+        d2 = Dashboard.objects.create(team=self.team, name="b dashboard", created_by=self.user)
+        d3 = Dashboard.objects.create(team=self.team, name="c dashboard", created_by=self.user)
+
+        # Makes 8 queries right now
+        # For permissions: session, user, team, organization membership, organization
+        # For dashboards: dashboard count, dashboard list, dashboard items
+        with self.assertNumQueries(8):
+            response = self.client.get(f"/api/projects/{self.team.id}/dashboards/")
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            response_data = response.json()
+            self.assertEqual(len(response_data["results"]), 3)
```

Comment:
Can we just do one assert here, checking that the array of names is as expected. Testing the len the the contents would be redundant and we’d get better output on failures